samza-core/src/test/java/org/apache/samza/storage/TestTaskSideInputStorageManager.java

```diff
@@ -182,6 +186,36 @@ public void testGetFileOffsets() {
       });
   }
 
+  /**
+   * This test is for cases, when calls to systemAdmin (e.g., KafkaSystemAdmin's) get-stream-metadata method return null.
+   */
+  @Test
+  public void testGetStartingOffsetsWhenStreamMetadataIsNull() {
+    final String storeName = "test-get-starting-offset-store";
+    final String taskName = "test-get-starting-offset-task";
+
+    Set<SystemStreamPartition> ssps = IntStream.range(1, 6)
+        .mapToObj(idx -> new SystemStreamPartition("test-system", "test-stream", new Partition(idx)))
+        .collect(Collectors.toSet());
+    Map<Partition, SystemStreamMetadata.SystemStreamPartitionMetadata> partitionMetadata = ssps.stream()
+        .collect(Collectors.toMap(SystemStreamPartition::getPartition,
+            x -> new SystemStreamMetadata.SystemStreamPartitionMetadata(null, "1", "2")));
```

Comment:
What's the purpose for multiple ssps if they are each testing the same condition?

Reply:
the task-sideinputstoragemanager uses maps to index ssps so we should have more than 1, but youre right two are probably enough.